### PR TITLE
feat: example associate sessions with backend events

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

Periodically people get stuck with this... "I have backend events and I want to filter for sessions that contain them"

## Changes

The backend events that are added aren't super useful, they're simply there to show an example of how we can use the session_id from the header

Screenshot 2023-07-07 at 14 14 20

* adds session id as a header to every call made by `api.ts`
* uses that in session recording to show how it works

## How did you test this code?

👀 locally

### CODE REVIEW

**Comment by pauldambra** [comment]
needs https://github.com/PostHog/posthog-js/pull/727 to stop tests from freezing

File: frontend/src/lib/api.ts [hunk 1]
```diff
@@ -1430,6 +1436,7 @@ const api = {
             headers: {
                 ...(isFormData ? {} : { 'Content-Type': 'application/json' }),
                 'X-CSRFToken': getCookie(CSRF_COOKIE_NAME) || '',
+                // 'X-POSTHOG-SESSION-ID': posthog.get_session_id(),
```
Comment (daibhin) [inline frontend/src/lib/api.ts:1439]: Should these headers be commented out?

Comment (benjackwhite) [inline frontend/src/lib/api.ts:1439]: ??

**Review by benjackwhite — COMMENTED** [review]
I like it as an example. I wonder if there isn't a way that we could automatically add it like Flask's global thead-based thing where you can grab the current request if there is one...

But generally speaking - yeah looks good

File: frontend/src/lib/api.ts (hunk 1 above)
Comment (pauldambra) [reply frontend/src/lib/api.ts:1439]: Well done, you both spotted my deliberate mistake 🏅 

baby-circle

**Comment by pauldambra** [comment]
> I wonder if there isn't a way that we could automatically add it like Flask's global thead-based thing where you can grab the current request if there is one

I wondered about this as a middleware... I might be being slow... I couldn't think of a way to intercept the interaction with posthog-python

**Comment by neilkakkar** [comment]
Doesn't seem easy to intercept, buuut if you could pass the current (generic) request object to `posthoganalytics.capture()`, it could parse all the headers, figure out which ones are relevant to posthog, and add those as session props automagically.

This way, any future headers don't need custom handling in their backend, they just get added as props.

The more I think about this, the more I like this.

**Pull request merged**